lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,9 +166,64 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
-    return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
+
+  if (ppResult == nullptr || *ppResult == nullptr)
+    return S_OK;
```

Comment:
We should have tested `ppResult` and returned `E_INVALIDARG` if nullptr at the beginning of the function.
For `*ppResult == nullptr`, that indicates a failure from `DxcResult::Create`, which should have been caught already.

With the current code, for either of these conditions, the `DxcResult::Create` should have failed causing a return under `CATCH_CPP_RETURN_HRESULT()` before getting here.